asset/s3_exporter/processor.js

```diff
@@ -99,12 +106,25 @@ class S3Batcher extends BatchProcessor {
             throw new Error(`Unsupported output format "${this.opConfig.format}"`);
         }
 
+        // This will prevent empty objects from being added to the S3 store, which can cause
+        // problems with the S3 reader
+        if (outStr.length === 0) {
+            return Promise.resolve();
+        }
+
         const params = {
             Bucket: this.opConfig.bucket,
-            Key: objName,
-            Body: outStr
+            Key: objName
         };
 
+        if (this.opConfig.compression === 'lz4') {
+            params.Body = lz4.encode(outStr);
+        } else if (this.opConfig.compression === 'gzip') {
+            params.Body = await gzip(outStr);
+        } else {
+            params.Body = outStr;
+        }
+
         return this.client.putObject_Async(params)
             .then((results) => DataEntity.fromBuffer(JSON.stringify(results)));
```

Comment:
Returning the results isn't really useful here. I would probably just return the input slice (so this can be more composable).

Reply:
fixed